### ISSUE TITLE
chainer faster RCNN seems load pretrained model npz not success!

### ISSUE THREAD

**Issue by sharpstill** [chainer/chainercv#334]
After whole 2 days of training faster RCNN ( I have slight modify your code to meet my needs), I found that result/snapshot_model.npz  seems not load success.
I use https://github.com/chainer/chainercv/blob/master/chainercv/links/model/faster_rcnn/faster_rcnn_vgg.py#L139 to load npz file.
Today I wake up and to continue to train the model, then load the result/snapshot_model.npz, then I found in the start training , the loss is above 2.0, because in yestoday's training the loss has already drop to 0.5.
So I think this may be issue or bug inside chainer? 
But I found another interesting thing, The loss decrease fast and after 10~20 iterations catch up with yestoday 0.5.
Is this normal?

**Comment by sharpstill** [comment]
weird problem. look at below log:
```
iteration   epoch       elapsed_time  lr          main/loss   validation/main/all
20          0           20.9103       0.0001      1.17345                          
40          0           39.3119       0.0001      0.848364                         
60          0           57.7876       0.0001      0.669107                         
80          0           76.0707       0.0001      0.725216                         
100         0           96.1029       0.0001      0.57862180  
```
It decrease quickly and begin with 1.17, but yestoday I already decrease loss about equals 0.5
If I don't load .npz file , It will begin with 2.1, So I think it load npz may success, but because your class **FasterRCNN is inherit and composite of other 2 class head and vgg feature extractor class.**
**Does npz success load into other 2 component of FasterRCNN?**

**Comment by yuyu2172** [comment]
If you want to completely restart your training, you need to resume trainer's parameters as well. They contain optimizer params that are related to training.

You can check that in MNIST example.
You need to add an extension to save snapshot of the trainer and an extension to load the saved snapshot. These two are not included in the training code of Faster R-CNN.

**Comment by yuyu2172** [comment]
By the way, I am pretty sure that FasterRCNN is loading head, feature extractor and RPN properly.

**Comment by sharpstill** [comment]
@yuyu2172  I have figured out reason. after I load optimizer parameter from npz file, It works normally

**Comment by yuyu2172** [comment]
It is good to hear that.
